shared/src/main/java/uk/gov/di/authentication/shared/helpers/EmailCheckMetricsHelper.java

```diff
@@ -0,0 +1,43 @@
+package uk.gov.di.authentication.shared.helpers;
+
+import uk.gov.di.authentication.shared.entity.JourneyType;
+import uk.gov.di.authentication.shared.entity.NotificationType;
+import uk.gov.di.authentication.shared.entity.cloudwatch.CredentialType;
+import uk.gov.di.authentication.shared.services.CloudwatchMetricsService;
+
+import java.util.List;
+import java.util.Map;
+import java.util.Objects;
+
+import static uk.gov.di.authentication.shared.domain.CloudwatchMetricDimensions.CREDENTIAL_TYPE;
+import static uk.gov.di.authentication.shared.domain.CloudwatchMetricDimensions.ENVIRONMENT;
+import static uk.gov.di.authentication.shared.domain.CloudwatchMetricDimensions.JOURNEY_TYPE;
+import static uk.gov.di.authentication.shared.domain.CloudwatchMetrics.USER_SUBMITTED_CREDENTIAL;
+
+public class EmailCheckMetricsHelper {
+    private EmailCheckMetricsHelper() {}
+
+    private static final List<String> notificationSetupTypes =
+            List.of(
+                    NotificationType.VERIFY_EMAIL.name(),
+                    NotificationType.VERIFY_PHONE_NUMBER.name());
+
+    public static void incrementUserSubmittedCredentialIfNotificationSetupJourney(
+            CloudwatchMetricsService metricsService,
+            JourneyType journeyType,
+            String notificationType,
+            String environment) {
+        if (notificationSetupTypes.contains(notificationType)) {
+            var credentialType =
+                    Objects.equals(notificationType, NotificationType.VERIFY_EMAIL.name())
+                            ? CredentialType.EMAIL_ADDRESS
+                            : CredentialType.PHONE_NUMBER;
```

Comment:
As we may find any random string is put in as the `notificationType` in the future, could we assert the string is either `VERIFY_EMAIL` or `VERIFY_PHONE_NUMBER` before using it?

I'm aware if we make a mistake in the future that we may fallback to miss-recording a the `CredentialType` dimension as `PHONE_NUMBER`.

Reply:
Redacted, you are already doing this with `notificationSetupTypes.contains(notificationType)` in `EmailCheckMetricsHelper